app/v2/controllers/V2ArrivalsController.scala

```diff
@@ -79,21 +84,50 @@ class V2ArrivalsControllerImpl @Inject() (
         (for {
           _ <- validationService.validateXml(MessageType.ArrivalNotification, request.body).asPresentation
           _ = auditService.audit(AuditType.ArrivalNotification, request.body, MimeTypes.XML)
-
-          arrivalNotificationResult <- persistAndSend(request.body)
-        } yield arrivalNotificationResult).fold[Result](
+          arrivalResult <- arrivalsService.createArrival(request.eoriNumber, request.body).asPresentation
+          _ <- routerService
+            .send(MessageType.ArrivalNotification, request.eoriNumber, arrivalResult.arrivalId, arrivalResult.messageId, request.body)
+            .asPresentation
+        } yield arrivalResult).fold[Result](
           presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
           result => Accepted(HateoasArrivalNotificationResponse(result.arrivalId))
         )
     }
 
-  private def persistAndSend(
-    source: Source[ByteString, _]
-  )(implicit hc: HeaderCarrier, request: AuthenticatedRequest[Source[ByteString, _]]) =
+  def createArrivalNotificationJSON(): Action[Source[ByteString, _]] =
+    (authActionNewEnrolmentOnly andThen messageSizeAction()).stream {
+      implicit request =>
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
+        handleJson(MessageType.ArrivalNotification, request.body)
+          .flatMap {
+            src =>
+              withReusableSource(src) {
+                xmlSource =>
+                  for {
+                    _ <- validationService
+                      .validateXml(MessageType.ArrivalNotification, xmlSource)
+                      .asPresentation(jsonToXmlValidationErrorConverter, materializerExecutionContext)
+                    arrivalResult <- arrivalsService.createArrival(request.eoriNumber, xmlSource).asPresentation
+                    _ <- routerService
+                      .send(MessageType.ArrivalNotification, request.eoriNumber, arrivalResult.arrivalId, arrivalResult.messageId, xmlSource)
+                      .asPresentation
```

Comment:
I think we should keep `persistAndSend` as it was to avoid the duplication, because the source you'll pass will still work.